### PR TITLE
LibWeb: Add internal use_pseudo_selector to element

### PR DESCRIPTION
**Pull request by bplaat**
This pr changes the way internal elements that use a pseudo selector for there styles work. Currently a element needs to extend `DOM::Element` and implement the `pseudo_element()` virtual. This is quite clunky and results in many small class defines. I changed this to a `set_use_pseudo_selector()` function that can be applied to any element.

In `TreeBuilder.cpp` the psuedo selector is then used and the inline styles are merged back. This results in a clean base where more internal shadow dom elements that use pseudo selector can easily be added. Before some large `if`s checked this but with the setter and the inline style merge this is not needed any more.

First I wanted to add a new `psuedo` attribute like Chrome uses but I could not protect it to internal shadow doms only. So this internal element setter is the next best thing and also type complete!

TL;DR: This pr simplifies the way internal elements can force a pseudo selector on them self

Love to here your feedback on this change

### CODE REVIEW

**Comment by AtkinsSJ** [comment]
I've only given this a quick look over so far, but reducing the boilerplate is great!

I do see you using the phrase "pseudo selector" though. There are two kinds of those: pseudo-elements and pseudo-classes, which behave differently. We only care about pseudo-elements here, so please use that term to refer to them, so it's less confusing. :^)

**Comment by awesomekling** [comment]
Yeah that's much nicer :)
